src/common/tests_dmg_helpers.c

```diff
@@ -260,6 +260,10 @@ parse_pool_info(struct json_object *json_pool, daos_mgmt_pool_info_t *pool_info)
 		return -DER_INVAL;
 	}
 	uuid_str = json_object_get_string(tmp);
+	if (uuid_str == NULL) {
+		D_ERROR("unable to extract UUID string from JSON\n");
+		return -DER_INVAL;
+	}
 	uuid_parse(uuid_str, pool_info->mgpi_uuid);
```

Comment:
This line should be checked as well.

Reply:
uuid_parse uses uuid_str without checking NULL:
strlen() doesn't work with NULL point.

int uuid_parse(const char *in, uuid_t uu)
{
        struct uuid     uuid;
        int             i;
        const char      *cp;
        char            buf[3];

        if (strlen(in) != 36)
                return -1;

Reply:
What I meant was we should be checking the return code of uuid_parse, there could be a strong set in the yaml file which wasn't a valid uuid and we'd miss a chance to catch it here.

Reply:
done.